src/Components/UserProfile/ProfilePage.js

```diff
@@ -1,20 +1,38 @@
-import React from 'react'
+import React, { useEffect, useState } from 'react'
 import PropTypes from 'prop-types'
 
 import Profile from '../Profile'
 import Links from '../Links'
 import Milestones from '../Milestones'
 
 function ProfilePage({ profile, username }) {
+  const [links, setLinks] = useState([])
+  const [milestones, setMilestones] = useState([])
+
+  useEffect(() => {
+    if (profile.links) {
+      const newLinks = profile.links.map((link) => {
+        return { ...link, icon: link.icon?.toLowerCase() }
+      })
+      setLinks(newLinks)
+    }
```

Comment:
Not sure the check is needed because it is an empty array?

I think it will be cleaner to do something like this

```ts
const newLinks = profile.links.map((link) => ({ ...link, icon: link.icon?.toLowerCase()}));
```

Reply:
Yup , it's not needed if the list data are correctly added 

I found some profiles without milestones  and gave an undefined.
so I thought why take risk just add the check for links as well 🙂